### DISCUSSION TITLE
need help migrating content

### DISCUSSION THREAD

**Discussion by bbontrager-arcm** [PiranhaCMS/piranha.core#2065] [Q&A] [answered]
We are updating between major versions, Piranha 4.2 to 11.  Blocks were introduced along the way and we need to understand how to get our existing data stored in pages and fields to appear pages blocks .  Is there a migration script for creating the new database relationships between pages and our content?  Alternatively, is there an ERD for how all the new blocks/content/fields relate?  (With an ERD we could figure out the migration .)

**Comment by tidyui** [comment]
That’s one hell of a migration 🙂 I don’t think blocks should be an issue as it is a feature that was added on top of the features available before, ie blocks didn’t replace anything.

this means that after update your pages should retain their existing content but simply not have any blocks. Moving this content over to blocks means re-building the content, there’s no migration as it’s two completely different things.

If you are planning to re-build the content with blocks anyway it’s probably faster to start from scratch an move stuff over from the old project to the new bit by bit instead of trying to upgrade the existing project. An upgrade would feel logical if you wanted to retain all of the content as is.

**Reply by tidyui** [reply]
And btw, thank you so much for using Piranha for six years and coming back at to upgrade your project. That’s the best review anyone can get 👍🏼 we are happy to help out if you provide more details regarding your project, which parts you want to migrate vs recreate using the new features available!

**Reply by bbontrager-arcm** [reply]
Thanks!

Piranha provides our application a blog-like store of documentation (user guides and other supporting information for end-users), and also holds content that is shown on the landing page.  Our symptom is after updating we see pages and posts defined in the Manager, but no content appears in them on the landing page, or when we edit them.  If I understand you correctly, we should be able to get the original content (Pages and PageFields, or Posts and PostFields) to display without any data changes?  Perhaps instead of looking at the database, there is something we need to refactor in .NET?

**Comment by bbontrager-arcm** [comment, marked as answer]
This pointed us in the right direction.  

We needed to apply UseBlocks = false and/or IsArchive = true in a few places in our Model Templates.  Once we did that, the content in the Manager appeared as we are used to.

`[PageType(Title = "Blog archive", UseBlocks = false, IsArchive = true)]`

No content appeared on our landing page because options.UseCms() had not been added at startup.  We have some more debugging to do, but we can tell Piranha is trying to render the view now instead of the application's default html.
